src/components/SRNEntry.vue

```diff
@@ -78,49 +89,73 @@ export default {
       return this.redirectTo == "plio";
     },
     isSubmitButtonDisabled() {
-      return !this.isAnyUserIDPresent || this.invalidInputMessage != "";
+      //submit button is disabled if no SRN has been typed or input is invalid or SRN hasn't been completely typed
+      return (
+        !this.isAnyUserIDPresent ||
+        this.invalidInputMessage != "" ||
+        this.userIDList.slice(-1)[0]["userID"].length < this.maxLengthOfSRN - 1
+      );
     },
     isUserValidated() {
       //if multiple SRN's are allowed, then this helps with the activation of the + button, to add more SRN's
       return (
         !this.isSingleEntryOnly &&
-        this.userIDList[0]["userID"].length > this.maxLengthOfSRN - 1
+        this.userIDList.slice(-1)[0]["userID"].length > this.maxLengthOfSRN - 1 &&
+        this.userIDList.length < 10
       );
     },
   },
   methods: {
     isValidSRNFormat(e) {
```

Comment:
```suggestion
    allowNumericEntriesOnly(e) {
```
As this is a method, the naming should be such that it tells us what the function does.